library/x509_crt.c

```diff
@@ -393,7 +393,7 @@ static int x509_get_version( unsigned char **p,
             return( 0 );
         }
 
-        return( ret );
+        return( MBEDTLS_ERR_X509_INVALID_FORMAT + ret );
```

Comment:
Is it guaranteed this won't ever overflow? Do we want to use `+` or `|`?

Reply:
Yes, we're returning an ASN.1 low level error code here, and wrapping it by a high-level X.509 error code. This is the idiom that we have throughout the library. As mentioned elsewhere, it's not optimal, because what _we_ know the compiler doesn't, namely that the error codes of X.509 and ASN.1 are binary disjoint; using `|` instead of `+` would at least allow the compiler or static analyzers to realize that the wrapping doesn't lead to zero. However, as mentioned, it's a common idiom throughout the library, so it should be changed everywhere or nowhere, but not in a single place.

Reply:
OK. Not that happy we are justifying not doing things the best we can in new code because we didn't in old code, but fine.

Reply:
I think it's better to know we're always _consistently_ using a particular idiom, even if it's not optimal and could be improved, than to mix idioms and loose track of where we already switched and where we didn't. Don't you think? I created https://github.com/ARMmbed/mbedtls/issues/2462.